pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -577,28 +604,136 @@ func (r *SubnetPortReconciler) addressBindingMapFunc(ctx context.Context, obj cl
 		}
 	}
 	log.Info("No SubnetPort found for AddressBinding", "namespace", ab.Namespace, "name", ab.Name, "VM", ab.Spec.VMName)
+	setAddressBindingStatus(r.Client, ctx, ab, metav1.Now(), vmOrInterfaceNotFoundError, "")
 	return nil
 }
 
-func setAddressBindingStatus(client client.Client, ctx context.Context, subnetPort *v1alpha1.SubnetPort, subnetPortService *subnetport.SubnetPortService) {
+func (r *SubnetPortReconciler) collectAddressBindingGarbage(ctx context.Context, namespace, ipAddress *string) {
+	abList := &v1alpha1.AddressBindingList{}
+	var err error
+	if namespace == nil || *namespace == "" {
+		err = r.Client.List(context.TODO(), abList)
```

Comment:
Why not use `ctx` instead of `context.TODO()`?

Reply:
Fixed.